README.md

```diff
@@ -29,95 +33,319 @@ are well established for RSE-ops, and which require further development. We can
+> lots of different machines IF we had test resources managed differently. Right now there are
+> still some roadblocks around service user accounts, banks (time allocations), and access to
+> machines that need to get worked out. Also most HPC sites won't have every architecture (we don't
+> even have everything -- just close to it) so building versions to distribute to everyone is hard.
+
+When software testing isn't in the hands of the user, testing the software stack might fall in the hands of the system administrator or support staff of the HPC cluster.
+
+> or nobody
+
 System administrators might use regression testing, including testing file permissions and mounts,
 communication between nodes and compatibility with the resource. For these regression tests, tools like Pavilion, NHC and Interstate are popular.
 
-Another continuous integration strategy is to focus on regression testing via tools like ReFrame. 
-Tools of this type try to test HPC systems alongside software stacks directly on the resource, and typically separate from any web interface or alongside the code. (**TODO** add more about ReFrame and other tools, I don't know a lot about it/them). The biggest difference between these two spaces is likely how software is deployed. On HPC the software is optimally installed natively, and in cloud native / DevOps it's more common to deploy containers and then run via an orchestrator or cloud service. For HPC, it's up to workload managers to run the software. These resources are also typically protected, and not easily accessible from a web interface. The challenge is thus getting any kind of modern web-based interface linked up directly to a cluster to run tests. Work at the lab with GitLab, and specifically having a locally deployed Gitlab server with custom runners on various resources has been a way to unite these two worlds.
+Another continuous integration strategy is to focus on regression testing via tools like ReFrame.
+Tools of this type try to test HPC systems alongside software stacks directly on the resource, and typically separate from any web interface or alongside the code. (**TODO** add more about ReFrame and other tools, I don't know a lot about it/them). The biggest difference between these two spaces is likely how software is deployed.
+
+> Big difference with ReFrame and Pavilion vs. (missing) app testing tools is that the former are
+> designed to test the *machine*, not the software. They run benchmarks to ensure the hardware
+> works, not the other way around. There is a lot of overlap and maybe potential here but an app
+> testing tool is likely tied to the software repo in a way that ReFrame is not... it would be nice
+> to see some convergence here.
+
+On HPC the software is optimally installed natively, and in cloud native / DevOps it's more common to deploy containers
+
+> I want to get away from the notion that containers are not "native" -- they can be built and
```

Comment:
directly on the system?

Reply:
"bare metal" is the thing I hear frequently.